### ISSUE TITLE
False positive SC2094

### ISSUE THREAD

**Issue by 0xPaul** [koalaman/shellcheck#2562]
#### For bugs
- SC2094
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2094)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
mapfile -t foo <foo
```

#### Here's what shellcheck currently says:
Make sure not to read and write the same file in the same pipeline.
See SC2094.

#### Here's what I wanted or expected to see:
Nothing 🙂. `mapfile` creates a variable, not a file.
